editoast/src/views/timetable/stdcm.rs

```diff
@@ -638,7 +733,205 @@ async fn parse_stdcm_steps(
+          "time_gap_before": 35000
+        })
+    }
+
+    fn core_mocking_client() -> MockingClient {
+        let mut core = MockingClient::new();
+        core.stub("/v2/pathfinding/blocks")
+            .method(reqwest::Method::POST)
+            .response(StatusCode::OK)
+            .json(
+                serde_json::to_value(PathfindingResult::Success(pathfinding_result_success()))
+                    .unwrap(),
+            )
+            .finish();
+        core.stub("/v2/standalone_simulation")
+            .method(reqwest::Method::POST)
+            .response(StatusCode::OK)
+            .json(serde_json::to_value(simulation_response()).unwrap())
+            .finish();
+        core
+    }
+
+    #[rstest]
+    async fn stdcm_return_success() {
+        let db_pool = DbConnectionPoolV2::for_tests();
+        let mut core = core_mocking_client();
+        core.stub("/v2/stdcm")
+            .method(reqwest::Method::POST)
+            .response(StatusCode::OK)
+            .json(json!({
+                "status": "success",
+                "simulation": serde_json::to_value(simulation_response()).unwrap(),
+                "path": serde_json::to_value(PathfindingResult::Success(pathfinding_result_success())).unwrap(),
+                "departure_time": "2024-09-17T20:05:00+02:00"
+            }))
+            .finish();
+
+        let app = TestAppBuilder::new()
+            .db_pool(db_pool.clone())
+            .core_client(core.into())
+            .build();
+        let small_infra = create_small_infra(&mut db_pool.get_ok()).await;
+        let timetable = create_timetable(&mut db_pool.get_ok()).await;
+        let rolling_stock =
+            create_fast_rolling_stock(&mut db_pool.get_ok(), &Uuid::new_v4().to_string()).await;
+
+        let request = app
+            .post(format!("/timetable/{}/stdcm?infra={}", timetable.id, small_infra.id).as_str())
+            .json(&stdcm_payload(rolling_stock.id));
+
+        let stdcm_response: STDCMResponse =
+            app.fetch(request).assert_status(StatusCode::OK).json_into();
+        assert_eq!(
+            stdcm_response,
+            STDCMResponse::Success {
+                simulation: simulation_response(),
+                path: pathfinding_result_success(),
+                departure_time: DateTime::from_str("2024-09-17T18:05:00Z")
+                    .expect("Failed to parse datetime")
+            }
+        );
+    }
+
+    #[rstest]
+    async fn stdcm_return_conflicts() {
+        let db_pool = DbConnectionPoolV2::for_tests();
+        let mut core = core_mocking_client();
+        core.stub("/v2/stdcm")
+            .method(reqwest::Method::POST)
+            .response(StatusCode::OK)
+            .json(json!({"status": "path_not_found"}))
+            .finish();
+        core.stub("/v2/conflict_detection")
+            .method(reqwest::Method::POST)
+            .response(StatusCode::OK)
+            .json(json!({"conflicts": []}))
```

Comment:
Might be interesting to add a conflict, just to see if we forward it correctly.

Reply:
Done.